common/src/main/java/io/pravega/common/util/ToStringUtils.java

```diff
@@ -63,4 +74,43 @@
                   .collect(toMap(e -> keyMaker.apply(e.getKey()), e -> valueMaker.apply(e.getValue())));
     }
 
+    /**
+     * Convert the given string to its compressed base64 representation.
+     * @param string String to be compressed to base64.
+     * @return String Compressed Base64 representation of the input string.
+     * @throws IOException If an I/O exception occurs.
+     * @throws NullPointerException If string is null.
+     */
+    public static String compressToBase64(final String string) throws IOException {
+        Preconditions.checkNotNull(string, "string");
+        final byte[] bytes;
+        @Cleanup
+        final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        @Cleanup
+        final GZIPOutputStream gzipOutputStream = new GZIPOutputStream(byteArrayOutputStream);
+        gzipOutputStream.write(string.getBytes(StandardCharsets.UTF_8));
+        gzipOutputStream.close();
+        bytes = byteArrayOutputStream.toByteArray();
+
+        return Base64.getEncoder().encodeToString(bytes);
+    }
+
+    /**
+     * Get the original string from its compressed base64 representation.
+     * @param base64CompressedString Compressed Base64 representation of the string.
+     * @return The original string.
+     * @throws IOException If an I/O exception occurs.
+     * @throws NullPointerException If base64CompressedString is null.
+     * @throws IllegalArgumentException If base64CompressedString is not null, but has a length of zero or if the string has illegal base64 character.
+     */
+    public static String decompressFromBase64(final String base64CompressedString) throws IOException {
+        Exceptions.checkNotNullOrEmpty(base64CompressedString, "base64CompressedString");
+        byte[] dataBytes = Base64.getDecoder().decode(base64CompressedString);
```

Comment:
From what I see, `getDecoder` and `getEncoder` can wrap existing `InputStreams`, which (I hope) will save you from having to allocate the extra array, and thus passing through it twice.

Please check it out and, if so, change it both here and in `compressToBase64` above.

Reply:
Yes, wrapping a stream with `getDecoder().wrap(..)` and `getEncoder().wrap(..)` is elegant and avoids using an extra array.
Updated the `compressToBase64` and `decompressFromBase64` to use it.